reposilite-plugins/prometheus-plugin/src/main/kotlin/com/reposilite/plugin/prometheus/PrometheusPlugin.kt

```diff
@@ -32,38 +41,77 @@ class PrometheusPlugin : ReposilitePlugin() {
+                reposiliteMetrics.responseFileSizeSummary.observe(info.contentLength.toDouble())
+                reposiliteMetrics.resolvedFileCounter.inc()
+            }
+        }
+
+        event { _: DeployEvent ->
+            reposiliteMetrics.mavenDeployCounter.inc()
+        }
+
+        logger.info("Prometheus | Reposilite metrics has been initialized")
+
+
         event { event: HttpServerConfigurationEvent ->
             val server = event.config.pvt.jetty.server!!
+            val handler = StatisticsHandler()
+            server.handler = handler
+
+            val jettyMetrics = JettyStatisticsHandlerMetrics.builder {
+                statisticsHandler = handler
+            }.register()
+
+            // Abusing request log, as it's the only way to get the time the request was sent
+            server.setRequestLog { request, response ->
+                jettyMetrics.responseTimeSummary.labelValues(response.status.toString())
+                    .observe((System.currentTimeMillis() - request.timeStamp).milliseconds.toDouble(DurationUnit.SECONDS))
+
+                jettyMetrics.responseSizeSummary.observe(response.contentCount.toDouble())
+            }
+
+            event.config.router.mount {
+                it.after { ctx ->
+                    jettyMetrics.responseCounter.labelValues(ctx.statusCode().toString()).inc()
+                }
+            }
+
+            logger.info("Prometheus | Jetty StatisticsHandler metrics has been initialized.")
 
-            val statisticsHandler = StatisticsHandler()
-            server.handler = statisticsHandler
-            StatisticsHandlerCollector.initialize(statisticsHandler)
-            logger.info("Prometheus | Default StatisticsHandler collector has been initialized.")
 
             when (val threadPool = server.threadPool) {
                 is QueuedThreadPool -> {
-                    QueuedThreadPoolCollector.initialize(threadPool)
-                    logger.info("Prometheus | QueuedThreadPool collector has been initialized.")
+                    QueuedThreadPoolMetrics.builder {
+                        queuedThreadPool = threadPool
+                    }.register()
+                    logger.info("Prometheus | Queued Thread Pool metrics has been initialized.")
                 }
-                else -> logger.info("Prometheus | Unsupported thread pool for collector: ${threadPool.javaClass.name}")
+
+                else -> logger.warn("Prometheus | Unsupported thread pool for metrics: ${threadPool.javaClass.name}, ignoring")
             }
         }
 
         event { event: RoutingSetupEvent ->
-            event.registerRoutes(
-                PrometheusEndpoints(
-                    prometheusFacade = prometheusFacade,
-                    prometheusPath = prometheusPath
-                )
-            )
+            event.registerRoutes(PrometheusEndpoints(prometheusFacade = prometheusFacade, prometheusPath = prometheusPath))
```

Comment:
nit: I don't think it was needed :sweat_smile:

Reply:
fair, I can revert that

I just try and compress lines when it doesn't increase clarity yk?